### ISSUE TITLE
💡 exclude individual tags from the report

### ISSUE THREAD

**Issue by GitHofi** [almarklein/timetagger#128]
The situation for which I am looking for a solution is the following:

In addition to the job/client, I also tag the activities by their "type". Was it a focused work, did I only do routine work or did I participate in a meeting etc.

For the " invoicing" of the hours I do not need these times. That means it would be useful if I could define a list of tags that are NOT included in the report. This could be done directly at the tag where you can also set a time target and the color → set a mark "don't include this tag in the report".

Do others have a similar situation?
Is there a better solution than my suggestion?

My suggestion would bring me a significant relief day by day, but also raises a new question: 
How can I then evaluate how much time I spent focused, in meetings, etc.?

Thanks
Chris

<img width="299" alt="2022_0126_1758_01" src="https://user-images.githubusercontent.com/12327228/151231385-7f372533-3b10-4e59-970f-2eacb63c3851.png">

**Comment by almarklein** [comment]
I see what you mean.

Thinking out loud, some options: 

1. Make it possible to mark tags as not-visible-in-report, as you suggested. Maybe the tag-target and tag-color dialogs can be merged into a tag-options dialog, to which a feature like this can be added.
2. Hide tags with a specific prefix / suffix. Nah, that's probably weird.
3. In the report dialog, allow hiding tags, e.g. by clicking on a cross behind their name. A downside would be that you'd have to do it every time. Unless there's a way to bring it back again. But then it looks an awful lot like (1).

I think (1) can work. I'll let this simmer for a bit. Other ideas are welcome :)

**Comment by GitHofi** [comment]
Hello,
I have another variant that would ease my problem:

💡 An option that allows to show all tags individually 
(and not the tag combinations) and their time sums in the report.

so instead of  ...
#kommunication #client1 2:00
#kommunication #client2 1:00

I would get ...
#kommunication 3:00
#client1 2:00
#client2 1:00

Then I could pick the hours that are relevant myself.

However, I think that the variant "mark tags as not-visible-in-report" would be the more flexible one and cover more use cases.
Anyway, the above variant would be an improvement to the is-state for me as well.

**Comment by almarklein** [comment]
A problem with showing each tag individually is that the numbers don't add up to the total, and in the report, some records would have to be shown twice (I think?), which would easily lead to confusion.

----

This issue somewhat relates to #120, which is now implemented: in the report dialog you can now choose to group by date (or by nothing) instead of tags. To what extend does this solve your use-case?

**Comment by GitHofi** [comment]
Thanks for those additional thoughts.
I agree, the single tag variant is probably more confusing than it helps.

Grouping by tags in the Overview (and Report) as a default view makes sense and is certainly a good default view.

It does NOT solve my problem though:

- #communication #projectA 2:00
- #communication #projectB 2:30
- #coding #projectA 3:00

... doesn't tell me how much time I have invested in projectA. Since I work on many activities (projects) in parallel, I have to gather the times individually.

(_By the way, I think the new grouping in the report is great_ 👍 )

**Comment by almarklein** [comment]
That makes sense. This issue still stands 👍

**Comment by almarklein** [comment]
Another idea, making it possible to mark a tag as either primary or secondary. This information can then also be used by the logic that sorts the tags in the overview (structure by primary tags only), and in the report dialog there could be a checkbox to hide secondary tags.

**Comment by peppobon** [comment]
Sorry, but I do not understand the issue. 
I too use subtags as in
    #communication #projectA 2:00
    #communication #projectB 2:30
    #coding #projectA 3:00
and I can already filter on #projectA  and then on #communication (or any conbination) to get the the relevant total hours spent.

Actually I use also a #non-billable tag to keep track of non billable hours.

It is just a matter of using a consistent tag vocabulary: I use a #client #project #activity-name #activity-type (i.e. #billable/#not-billable) structure and then I can have all the grouping I want.

BTW, the new grouping panel is super. Thanks Almar.

**Comment by almarklein** [comment]
The way I see it, this issue is because  a "sub-tag" is used so much that the overview is structured based on it, rather than on the "primary" tags (e.g. #clientA). Something like this:
```
#subtag #client1
#subtag #client2
#subtag #something-else
#client1 #meeting
#client1 #coding
```

This is expressed both in the overview panel, and in the report dialog. Both use the same [algorithm](https://github.com/almarklein/timetagger/blob/9f14c730875ded9eb854f88259b5d05f0f993bcb/timetagger/app/utils.py#L218-L308). 

In the overview it's not that much of a problem, because one can always see the totals by hoovering over the tag, or click the tag to get a breakdown for it.

But in the report you don't have this interactivity, and there is no way to affect the way things are ordered. Also, some sub-tags are too much detail (e.g. in the PDF to send to the client).

Earlier proposals only focused on the report dialog. My latest proposal is a more generic solution that would also make the overview more clear. There may still be other options though ...

**Comment by GitHofi** [comment]
> Sorry, but I do not understand the issue.

No problem - I'll try to describe it in more detail.

One aspect of this has already been described by almarklein:
> The way I see it, this issue is because a "sub-tag" is used so much that the overview is structured based on it, rather than on the "primary" tags. 

And then there is a second problem:
Yes, you can hover over the tags and see the total hours - that's how I do it now.
BUT there is a problem → one tag can appear many times in the overview - but I need the total hours only once. This is then doubly tedious. 
➊ First, you have to move the mouse over the tags to see the times and 
➋ second, you have to remember which tags you have already collected so you don't count them again when they appear a second or third time. And this problem is repeated anew with the other tags.
This is simply unnecessarily cumbersome.

This whole procedure to get the hours is no comparison to simply seeing the numbers in the report as you need them.

Thanks for the lively discussion. 
I am confident that we will find a good solution here.

**Comment by almarklein** [comment]
Working on it in #188!

**Comment by almarklein** [comment]
Released and deployed! 🍾
